### ISSUE TITLE
Parallel Compilation, Caching & Race Conditions

### ISSUE THREAD

**Issue by jswrenn** [brownplt/pyret-lang#1423]
One scenario is this: let's say we want to compile & run two files (_A.arr_ and _B.arr_) in parallel that share a dependency (e.g., `definitions.arr`); both compilation tasks may simultaneously attempt to cache the compiled result of `definitions.arr` in the `build` directory and the writes they perform may be interleaved and the cached file will be corrupted and syntactically-invalid.

**Comment by schanzer** [comment]
@jswrenn does this fix in powdermonkey resolve the issue in pyret-lang?

**Comment by jswrenn** [comment]
Not really; it just side-steps it by forcing each instance to use a different folder for caching.
